crates/noirc_evaluator/src/ssa_refactor/acir_gen/mod.rs

```diff
@@ -375,12 +375,16 @@ impl Context {
             .expect("Expected array index to fit into a u64") as usize;
 
         if index >= array.len() {
-            // Ignore the error if side effects are disabled.
             if let Some(var) = self.current_side_effects_enabled_var {
+                // Ignore the error if side effects are disabled.
                 if self.acir_context.is_constant_one(&var) {
                     // TODO: Can we save a source Location for this error?
                     panic!("Index {} is out of bounds for array of length {}", index, array.len());
                 }
+            } else {
+                // If `current_side_effects_enabled_var` are `None` it means
+                // that we are in the main function and want to panic
+                panic!("Index {} is out of bounds for array of length {}", index, array.len());
```

Comment:
Can we revert this change in favor of #1906? I'd rather remove the `Option` entirely.

Reply:
yup will revert

Reply:
done